compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -2184,20 +2184,39 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 let suffix = chars.collect::<String>();
                 suffix.is_empty() || suffix == "f32" || suffix == "f64"
             };
+            let is_likely_suffix = |first_chars: &[char], field: &str| -> Option<&str> {
+                if field.len() >= 1
+                    && field.to_lowercase().starts_with(first_chars)
+                    && field[1..].chars().all(|c| c.is_ascii_digit())
+                {
+                    if field.to_lowercase().starts_with(['f']) { Some("f32") } else { Some("f64") }
```

Comment:
With the addition of this condition, the `is_likely_suffix` is a bit less useful because it really only makes sense if `first_chars` is `f` or `l`, so maybe we could just do away with that parameter?

Reply:
Sounds reasonable.

Reply:
I'd probably rename the function to indicate its new specificity if making this change, like `maybe_partial_suffix`/`maybe_likely_suffix` or something?

Reply:
Updated it!